### PR TITLE
Update to .NET 7

### PR DESCRIPTION
**Pull request by Twinki14**
# Motivations
Maintenance 

# Modifications
- Update the tests projects to .NET 7
- Update `TinyHealthCheck` to .NET Standard 2.1
- Update to C# 11 lang
- Update various packages to latest
- Add notes about using `*` or `+` as the health check hostname instead of `localhost`, in .NET 7 for some reason `localhost` breaks on the HttpListener if you're running your app inside a docker container (quite frankly too lazy to research into why, but this is the suggested fix according to stock overflow https://stackoverflow.com/questions/75961828/c-sharp-net-core-7-http-listener-app-wrapped-in-docker-cannot-be-reached-from)

# Finally
Thank you for this library, this was *exactly* what I needed for a console application specifically using a generic host instead of a web host but still hosting it on a k8s cluster

### CODE REVIEW

**Comment by Twinki14** [comment]
@bruceharrison1984 Heyo! This is ready, I can't find any breaking changes, and quite frankly this isn't even needed to fix anything broken but I wanted to contribute in some way for such a brilliantly small lib

All this does is updated various versions, packages, apply some formatting here and there, and add some notes about docker containers

**Comment by bruceharrison1984** [comment]
I appreciate the PR! I'll give this a look over tomorrow and let you know if anything needs to be changed. This would likely be a candidate for a 1.0.0 release since it's been pretty extensively battle tested at this point.

File: .github/workflows/devBuild.yml
Comment (bruceharrison1984) [inline .github/workflows/devBuild.yml:1]: Can we avoid changing the indention and whitespace, thus flagging unnecessary changes.

File: .github/workflows/releaseBuild.yml
Comment (bruceharrison1984) [inline .github/workflows/releaseBuild.yml:1]: Can we avoid changing the indention and whitespace, thus flagging unnecessary changes.

File: TinyHealthCheck/TinyHealthCheck.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netstandard2.1</TargetFramework>
```
Comment (bruceharrison1984) [inline TinyHealthCheck/TinyHealthCheck.csproj:4]: This change actually carries some weight, and I am not a fan. I actually stopped at `netstandard2.0` to preserve backwards compatibility with .Net Framework. Moving to netstandard2.0 remove all .Net Framework compatibility, as well as several other implementations.

https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0#select-net-standard-version

I would ask that you put this back to `netstandard2.0` to keep the compatibility as high as possible.

**Review by bruceharrison1984 — CHANGES_REQUESTED** [review]

File: .github/workflows/devBuild.yml
Comment (Twinki14) [reply .github/workflows/devBuild.yml:1]: Apologies, it's been adjusted

File: .github/workflows/releaseBuild.yml
Comment (Twinki14) [reply .github/workflows/releaseBuild.yml:1]: Apologies, it's been adjusted

File: TinyHealthCheck/TinyHealthCheck.csproj (hunk 1 above)
Comment (Twinki14) [reply TinyHealthCheck/TinyHealthCheck.csproj]: Makes sense, revert back to 2.0

**Comment by Twinki14** [comment]
@bruceharrison1984 Ready for another review, I did also update the release workflow to use the newer `>> $GITHUB_OUTPUT`, so a minor additional update

**Comment by bruceharrison1984** [comment]
Merged. I'll cut a new release shortly. Thanks for the help!

**Pull request merged**